mongodb/src/test/java/org/hibernate/ogm/datastore/mongodb/test/query/nativequery/MongoDBSessionCLIQueryTest.java

```diff
@@ -598,4 +606,24 @@ public void testInsertMultipleWithNumberLongThenRemove() throws Exception {
 	protected Class<?>[] getAnnotatedClasses() {
 		return new Class[] { OscarWildePoem.class };
 	}
+
+	@Test
+	@TestForIssue(jiraKey = "OGM-1247")
+	public void testDistinctQuery() throws Exception {
+		try (OgmSession session = openSession()) {
+			Transaction transaction = session.beginTransaction();
+
+			String nativeQuery = "db." + OscarWildePoem.TABLE_NAME + ".distinct(\"name\",{\"author\":\"Oscar Wilde\"})";
+
+			Query query = session.createNativeQuery( nativeQuery );
+
+			List result = (List) session.createNativeQuery( nativeQuery ).uniqueResult();
+
+
+			assertThat( result.size() ).isEqualTo( 3 );
```

Comment:
A separate method, now you have `testDistinctQuery()`, you need to add `testDistinctQueryWithoutCriteria()` for example. See what we have done for Aggregate in the same class